app/models/competition.rb

```diff
@@ -397,6 +397,14 @@ def main_event
     Event.c_find(main_event_id)
   end
 
+  def report_posted?
+    delegate_report.posted?
+  end
+
+  def visible?
+    self.showAtAll
+  end
```

Comment:
I think `alias_attribute :visible, :showAtAll` might also work. Rails has automated `...?` helpers for boolean DB columns, and I would expect that these helpers even transfer across aliases.